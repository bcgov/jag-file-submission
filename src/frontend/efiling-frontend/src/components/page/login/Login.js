import React from "react";
import { Button } from "shared-components";
import PropTypes from "prop-types";
import Keycloak from "keycloak-js";

export default function Login({ keycloak }) {
  const authenticateWithKeycloak = (idp) => {
    sessionStorage.setItem("validExit", true);
    console.log(idp);
    keycloak.login({
      idpHint: `${idp}`,
    });
  };

  return (
    <div style={{ "text-align": "center", "line-height": "100px" }}>
      <Button
        label="Log in with BCeID"
        onClick={() => authenticateWithKeycloak("bceid")}
        styling={"bcgov-normal-blue btn"}
      />
      <Button
        label="Log in with IDIR"
        onClick={() => authenticateWithKeycloak("idir")}
        styling={"bcgov-normal-white btn"}
      />
      <Button
        label="Log in with BC Services Card"
        onClick={() => authenticateWithKeycloak("bcsc")}
        styling={"bcgov-normal-blue btn"}
      />
    </div>
  );
}

Login.propTypes = {
  keycloak: PropTypes.instanceOf(Keycloak).isRequired,
};
