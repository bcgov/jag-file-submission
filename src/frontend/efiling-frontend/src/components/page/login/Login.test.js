import React from "react";
import { render, waitFor } from "@testing-library/react";

import Login from "./Login";

const keycloak = jest.fn();

describe("Login Component", () => {
  test("BCeID button tries to log in with BCeID", async () => {
    const { asFragment } = render(<Login keycloak={keycloak} />);

    await waitFor(() => {});

    expect(asFragment()).toMatchSnapshot();
  });
});
