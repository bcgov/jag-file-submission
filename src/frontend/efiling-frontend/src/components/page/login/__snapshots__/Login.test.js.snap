// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component BCeID button tries to log in with BCeID 1`] = `
<DocumentFragment>
  <div
    style="text-align: center; line-height: 100px;"
  >
    <button
      class="bcgov-button bcgov-normal-blue btn"
      data-testid=""
      type="button"
    >
      Log in with BCeID
    </button>
    <button
      class="bcgov-button bcgov-normal-white btn"
      data-testid=""
      type="button"
    >
      Log in with IDIR
    </button>
    <button
      class="bcgov-button bcgov-normal-blue btn"
      data-testid=""
      type="button"
    >
      Log in with BC Services Card
    </button>
  </div>
</DocumentFragment>
`;
